go-controller/pkg/metrics/metrics.go

```diff
@@ -207,13 +214,27 @@ func StartOVNMetricsServer(bindAddress string) {
 	go utilwait.Until(func() {
 		err := http.ListenAndServe(bindAddress, mux)
 		if err != nil {
-			utilruntime.HandleError(fmt.Errorf("starting metrics server failed: %v", err))
+			utilruntime.HandleError(fmt.Errorf("starting ovn-metrics server failed: %v", err))
+		}
+	}, 5*time.Second, utilwait.NeverStop)
```

Comment:
in both of these functions you could use `utilwait.Forever()` rather than calling `utilwait.Until()` but passing `utilwait.NeverStop`